manager.go

```
@@ -611,6 +698,10 @@ func convertListener[T any](listener any) (Listener[T], int, error) {
 		return ConvertListener[M, T](lt), Normal, nil
 	case ListenerItem[M]:
 		return ConvertListener[M, T](lt.Listener), lt.Priority, nil
+	case func(Event[T]) error:
+		return NewListenerFunc(lt), Normal, nil
+	case *ListenerFunc[T]:
```

golangci-lint (1.64.8)

SA4020: unreachable case clause: go.lumeweb.com/event/v2.Listener[T] will always match before *go.lumeweb.com/event/v2.ListenerFunc[T]

(staticcheck)